File: src/main/java/com/practice/jblog/dto/SearchResult.java
package com.practice.jblog.dto;

import com.practice.jblog.Entity.SearchableEntity;

import java.util.List;
import java.util.Map;

public interface SearchResult <T extends SearchableEntity> {
    List<T> getResultItems();
    Map<String, Map<String, List<String>>> getHighlights();
    String getSearchResultsLink();
    long getResultsCount();
}

File: ts-frontend/src/models/SearchResult.ts
import Post from "./Post";

class SearchResult {
    constructor(
        public resultItems: Post[], 
        public highlights: Record<string, Record<string, string>>,
         public searchResultsLink: string, 
         public resultsCount: number) {

    }
   //  public get getResultItems(): Post[] {
   //      return this.resultItems;
   //   }

   //   public getHighlights(): string[] {
   //      return this._highlights;
   //   }

   //   public getSearchResultsLink(): string {
   //      return this._searchResultsLink;
   //   }

   //   public getResultsCount(): number {
   //      return this._resultsCount;
   //   }
}

export default SearchResult;